spring-cloud-skipper-server/src/test/java/org/springframework/cloud/skipper/server/service/ReleaseServiceTests.java

```diff
@@ -126,12 +126,55 @@ public void testStatusReleaseExist() {
 		packageIdentifier.setPackageName("log");
 		packageIdentifier.setPackageVersion("1.0.0");
 		installRequest.setPackageIdentifier(packageIdentifier);
-		releaseService.install(installRequest);
-		Info info = releaseService.status("testexists");
+		this.releaseService.install(installRequest);
+		Info info = this.releaseService.status("testexists");
 		assertThat(info).isNotNull();
 	}
 
 	@Test
+	public void testInstallReleaseThatIsNotDeleted() {
+		InstallProperties installProperties = new InstallProperties();
+		String releaseName = "installDeployedRelease";
+		installProperties.setReleaseName(releaseName);
+		installProperties.setPlatformName("default");
+		InstallRequest installRequest = new InstallRequest();
+		installRequest.setInstallProperties(installProperties);
+		PackageIdentifier packageIdentifier = new PackageIdentifier();
+		packageIdentifier.setPackageName("log");
+		packageIdentifier.setPackageVersion("1.0.0");
+		installRequest.setPackageIdentifier(packageIdentifier);
+		Release release = this.releaseService.install(installRequest);
+		assertThat(release).isNotNull();
+		try {
+			this.releaseService.install(installRequest);
+			fail("Expected to fail when installin already deployed release.");
+		}
+		catch (SkipperException e) {
+			assertThat(e.getMessage()).isEqualTo("Release with the name [" + releaseName + "] already exists "
+					+ "and it is not deleted.");
+		}
+	}
+
+	@Test
+	public void testInstallDeletedRelease() {
```

Comment:
Maybe it was out of scope for this PR, driven by the SCDF deploy an already deleted stream use-case, but I will add a test (and any changed) to support rolling back a deleted release.